portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -135,18 +149,27 @@ public void doDelete(HttpServletRequest request, HttpServletResponse response) t
 
         if (id > 0) {
             Key commentEntityKey = KeyFactory.createKey("Comment", id);
-            DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-            datastore.delete(commentEntityKey);
+
+            Query.Filter queryFilter = new Query.FilterPredicate("__key__", Query.FilterOperator.EQUAL, commentEntityKey);
+            query.setFilter(queryFilter);
+            PreparedQuery results = datastore.prepare(query);
+            Entity entity = results.asSingleEntity();
+
+            String commentUserId = (String) entity.getProperty("userId");
+            
+
+            // Only allows comments written by user logged in to be deleted
+            if (userId.equals(commentUserId)){
+                datastore.delete(commentEntityKey);
+            } 
         }
-    
     }
-    
-    response.setContentType("text/html");
-    response.getWriter().println();
 
-    // Redirect back to the HTML page.
-    response.sendRedirect("/index.html");
-  }
+    response.setContentType("text/html;");
 
+    // Redirect back to the jsp page.
+    response.getWriter().println();
+    response.sendRedirect("/");
```

Comment:
This is why the browser is requesting your root URL with the DELETE method and getting a 405 error; `fetch` automatically follows this redirect after getting the response from its DELETE /comments request.

Reply:
Ahhh, that makes sense. I'll remove the redirect to prevent this error. Thanks for troubleshooting this, I appreciate it!